### DISCUSSION TITLE
How to set direction for editor?

### DISCUSSION THREAD

**Discussion by strdr4605** [facebook/lexical#3810] [Q&A]
I have this example editor. Its direction is `null`

https://codesandbox.io/s/lexical-horizontal-rule-plugin-example-forked-kv2yht?file=/src/Editor.js:754-944

But the Editor from the playground has the direction set to `ltr`.

How is the direction set initially?

Also, is there a way to set the direction for all nodes, so that on node creation it will infer the correct direction?

Use case:
> We need to convert from ReactNative custom editor to Lexical editor state and we do this with lexical headless

--- 

Investigation:

I checked https://github.com/facebook/lexical/search?q=setDirection&type=code and looks like setDirection is used mostly when `static importJSON()`. 
But where is the main `this.setDirection('ltr')`?

**Comment by ssyberg** [comment]
Did you ever figure this out?

**Comment by ke-sn** [comment]
> How is the direction set initially?

After a quick investigation, I found it happen here
https://github.com/facebook/lexical/blob/f2b239ee7685d793a98ded23704a65dec6b043e7/packages/lexical/src/LexicalReconciler.ts#L406
